### ISSUE TITLE
IntelliJ 2022.3.x NullPointer on startup with ST4 plugin 0.9

### ISSUE THREAD

**Issue by jowilkes** [antlr/jetbrains-plugin-st4#42]
Issue in IntelliJ 2022.3.2 and some previous versions,  on IDE start this gets reported:

**Caused by: java.lang.NullPointerException: Cannot invoke "java.util.List.get(int)" because "org.antlr.jetbrains.st4plugin.psi.STTokenTypes.TOKEN_ELEMENT_TYPES" is null**

Not sure how they can be null.

# full trace:
```
com.intellij.diagnostic.PluginException: Cannot create class org.antlr.jetbrains.st4plugin.highlight.STBraceMatcher (classloader=PluginClassLoader(plugin=PluginDescriptor(name=StringTemplate v4, id=org.antlr.jetbrains.st4plugin, descriptorPath=plugin.xml, path=~/.local/share/JetBrains/Toolbox/apps/IDEA-U/ch-0/223.8617.56.plugins/jetbrains-plugin-st4, version=0.9, package=null, isBundled=false), packagePrefix=null, instanceId=171, state=active))
	at com.intellij.serviceContainer.ComponentManagerImpl.instantiateClass(ComponentManagerImpl.kt:989)
	at com.intellij.serviceContainer.ComponentManagerImpl.instantiateClass(ComponentManagerImpl.kt:1006)
	at com.intellij.serviceContainer.LazyExtensionInstance.createInstance(LazyExtensionInstance.java:55)
	at com.intellij.serviceContainer.LazyExtensionInstance.getInstance(LazyExtensionInstance.java:44)
	at com.intellij.serviceContainer.BaseKeyedLazyInstance.getInstance(BaseKeyedLazyInstance.java:38)
	at com.intellij.openapi.util.KeyedExtensionCollector.instantiate(KeyedExtensionCollector.java:177)
	at com.intellij.openapi.util.KeyedExtensionCollector.buildExtensionsFromExtensionPoint(KeyedExtensionCollector.java:164)
	at com.intellij.openapi.util.KeyedExtensionCollector.buildExtensions(KeyedExtensionCollector.java:139)
	at com.intellij.lang.LanguageExtension.buildExtensions(LanguageExtension.java:147)
	at com.intellij.lang.LanguageExtension.buildExtensions(LanguageExtension.java:16)
	at com.intellij.openapi.util.KeyedExtensionCollector.forKey(KeyedExtensionCollector.java:111)
	at com.intellij.lang.LanguageExtension.findForLanguage(LanguageExtension.java:101)
	at com.intellij.lang.LanguageExtension.forLanguage(LanguageExtension.java:93)
	at cn.aprilviolet.highlightbracketpair.highlighter.DefaultAbstractBracketHighlighter.refresh(DefaultAbstractBracketHighlighter.java:82)
	at cn.aprilviolet.highlightbracketpair.highlighter.DefaultAbstractBracketHighlighter.<clinit>(DefaultAbstractBracketHighlighter.java:38)
	at cn.aprilviolet.highlightbracketpair.highlighter.BracketHighlighterFactory.getBraceHighlighterInstance(BracketHighlighterFactory.java:42)
	at cn.aprilviolet.highlightbracketpair.component.HighlightEditorCartListener.highlightEditorCurrentPair(HighlightEditorCartListener.java:64)
	at cn.aprilviolet.highlightbracketpair.component.HighlightEditorCartListener.caretPositionChanged(HighlightEditorCartListener.java:54)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at com.intellij.util.EventDispatcher.dispatchVoidMethod(EventDispatcher.java:120)
	at com.intellij.util.EventDispatcher.lambda$createMulticaster$1(EventDispatcher.java:85)
	at jdk.proxy2/jdk.proxy2.$Proxy126.caretPositionChanged(Unknown Source)
	at com.intellij.openapi.editor.impl.CaretModelImpl.fireCaretPositionChanged(CaretModelImpl.java:543)
	at com.intellij.openapi.editor.impl.CaretImpl.doMoveToLogicalPosition(CaretImpl.java:461)
	at com.intellij.openapi.editor.impl.CaretImpl.moveToLogicalPosition(CaretImpl.java:606)
	at com.intellij.openapi.editor.impl.CaretImpl.lambda$moveToLogicalPosition$2(CaretImpl.java:318)
	at com.intellij.openapi.editor.impl.CaretModelImpl.doWithCaretMerging(CaretModelImpl.java:414)
	at com.intellij.openapi.editor.impl.CaretImpl.moveToLogicalPosition(CaretImpl.java:318)
	at com.intellij.openapi.editor.impl.CaretModelImpl.lambda$setCaretsAndSelections$4(CaretModelImpl.java:460)
	at com.intellij.openapi.editor.impl.CaretModelImpl.doWithCaretMerging(CaretModelImpl.java:419)
	at com.intellij.openapi.editor.impl.CaretModelImpl.setCaretsAndSelections(CaretModelImpl.java:447)
	at com.intellij.openapi.fileEditor.impl.text.TextEditorProvider.setStateImpl(TextEditorProvider.java:261)
	at com.intellij.openapi.fileEditor.impl.text.PsiAwareTextEditorProvider.setStateImpl(PsiAwareTextEditorProvider.java:101)
	at com.intellij.openapi.fileEditor.impl.text.AsyncEditorLoader.setEditorState(AsyncEditorLoader.kt:214)
	at com.intellij.openapi.fileEditor.impl.text.TextEditorImpl.setState(TextEditorImpl.java:163)
	at com.intellij.openapi.fileEditor.impl.FileEditorManagerImpl.restoreEditorState(FileEditorManagerImpl.java:1239)
	at com.intellij.openapi.fileEditor.impl.FileEditorManagerImpl.openFileImpl4Edt(FileEditorManagerImpl.java:1082)
	at com.intellij.openapi.fileEditor.impl.FileEditorManagerExImpl$openFileOnStartup$2.invokeSuspend$lambda$0(FileEditorManagerExImpl.kt:86)
	at com.intellij.openapi.application.TransactionGuardImpl.performActivity(TransactionGuardImpl.java:113)
	at com.intellij.openapi.fileEditor.impl.FileEditorManagerExImpl$openFileOnStartup$2.invokeSuspend(FileEditorManagerExImpl.kt:82)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at com.intellij.openapi.application.TransactionGuardImpl$1.run(TransactionGuardImpl.java:194)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:838)
	at com.intellij.openapi.application.impl.ApplicationImpl$3.run(ApplicationImpl.java:454)
	at com.intellij.openapi.application.impl.FlushQueue.doRun(FlushQueue.java:74)
	at com.intellij.openapi.application.impl.FlushQueue.runNextEvent(FlushQueue.java:114)
	at com.intellij.openapi.application.impl.FlushQueue.flushNow(FlushQueue.java:36)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:318)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:779)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:730)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:724)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:749)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:909)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:756)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$5(IdeEventQueue.java:437)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:787)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$6(IdeEventQueue.java:436)
	at com.intellij.openapi.application.TransactionGuardImpl.performActivity(TransactionGuardImpl.java:105)
	at com.intellij.ide.IdeEventQueue.performActivity(IdeEventQueue.java:615)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$7(IdeEventQueue.java:434)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:838)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:480)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:207)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:128)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:117)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:105)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:92)
Caused by: java.lang.ExceptionInInitializerError
	at java.base/jdk.internal.misc.Unsafe.allocateInstance(Native Method)
	at java.base/java.lang.invoke.DirectMethodHandle.allocateInstance(DirectMethodHandle.java:520)
	at com.intellij.serviceContainer.ComponentManagerImpl.instantiateClass(ComponentManagerImpl.kt:950)
	... 73 more
Caused by: java.lang.NullPointerException: Cannot invoke "java.util.List.get(int)" because "org.antlr.jetbrains.st4plugin.psi.STTokenTypes.TOKEN_ELEMENT_TYPES" is null
	at org.antlr.jetbrains.st4plugin.psi.STTokenTypes.getTokenElementType(STTokenTypes.java:23)
	at org.antlr.jetbrains.st4plugin.highlight.STBraceMatcher.<clinit>(STBraceMatcher.java:16)
	... 76 more
```

# runtime environment

IntelliJ IDEA 2022.3.2 (Ultimate Edition)
Runtime version: 17.0.5+1-b653.25 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.

Plugin version 0.9, installed via IntelliJ Marketplace from inside the IDE plugin management.
(Most recent entry in "What's New" relates to #37)

**Comment by jowilkes** [comment]
The induced.txt goes as follows:

Following exceptions happened soon after this one, most probably they are induced.


2023-03-10T09:21:30.209
com.intellij.diagnostic.PluginException: Cannot create class org.antlr.jetbrains.st4plugin.highlight.STBraceMatcher (classloader=PluginClassLoader(plugin=PluginDescriptor(name=StringTemplate v4, id=org.antlr.jetbrains.st4plugin, descriptorPath=plugin.xml, path=~/.local/share/JetBrains/Toolbox/apps/IDEA-U/ch-0/223.8836.35.plugins/jetbrains-plugin-st4, version=0.9, package=null, isBundled=false), packagePrefix=null, instanceId=171, state=active))
	at com.intellij.serviceContainer.ComponentManagerImpl.instantiateClass(ComponentManagerImpl.kt:989)
	at com.intellij.serviceContainer.ComponentManagerImpl.instantiateClass(ComponentManagerImpl.kt:1006)
	at com.intellij.serviceContainer.LazyExtensionInstance.createInstance(LazyExtensionInstance.java:55)
	at com.intellij.serviceContainer.LazyExtensionInstance.getInstance(LazyExtensionInstance.java:44)
	at com.intellij.serviceContainer.BaseKeyedLazyInstance.getInstance(BaseKeyedLazyInstance.java:38)
	at com.intellij.openapi.util.KeyedExtensionCollector.instantiate(KeyedExtensionCollector.java:177)
	at com.intellij.openapi.util.KeyedExtensionCollector.buildExtensionsFromExtensionPoint(KeyedExtensionCollector.java:164)
	at com.intellij.openapi.util.KeyedExtensionCollector.buildExtensions(KeyedExtensionCollector.java:139)
	at com.intellij.lang.LanguageExtension.buildExtensions(LanguageExtension.java:147)
	at com.intellij.lang.LanguageExtension.buildExtensions(LanguageExtension.java:16)
	at com.intellij.openapi.util.KeyedExtensionCollector.forKey(KeyedExtensionCollector.java:111)
	at com.intellij.lang.LanguageExtension.findForLanguage(LanguageExtension.java:101)
	at com.intellij.lang.LanguageExtension.forLanguage(LanguageExtension.java:93)
	at cn.aprilviolet.highlightbracketpair.highlighter.DefaultAbstractBracketHighlighter.refresh(DefaultAbstractBracketHighlighter.java:82)
	at cn.aprilviolet.highlightbracketpair.highlighter.DefaultAbstractBracketHighlighter.<clinit>(DefaultAbstractBracketHighlighter.java:38)
	at cn.aprilviolet.highlightbracketpair.highlighter.BracketHighlighterFactory.getBraceHighlighterInstance(BracketHighlighterFactory.java:42)
	at cn.aprilviolet.highlightbracketpair.component.HighlightEditorCartListener.highlightEditorCurrentPair(HighlightEditorCartListener.java:64)
	at cn.aprilviolet.highlightbracketpair.component.HighlightEditorCartListener.caretPositionChanged(HighlightEditorCartListener.java:54)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at com.intellij.util.EventDispatcher.dispatchVoidMethod(EventDispatcher.java:120)
	at com.intellij.util.EventDispatcher.lambda$createMulticaster$1(EventDispatcher.java:85)
	at jdk.proxy2/jdk.proxy2.$Proxy127.caretPositionChanged(Unknown Source)
	at com.intellij.openapi.editor.impl.CaretModelImpl.fireCaretPositionChanged(CaretModelImpl.java:543)
	at com.intellij.openapi.editor.impl.CaretImpl.doMoveToLogicalPosition(CaretImpl.java:461)
	at com.intellij.openapi.editor.impl.CaretImpl.moveToLogicalPosition(CaretImpl.java:606)
	at com.intellij.openapi.editor.impl.CaretImpl.lambda$moveToLogicalPosition$2(CaretImpl.java:318)
	at com.intellij.openapi.editor.impl.CaretModelImpl.doWithCaretMerging(CaretModelImpl.java:414)
	at com.intellij.openapi.editor.impl.CaretImpl.moveToLogicalPosition(CaretImpl.java:318)
	at com.intellij.openapi.editor.impl.CaretModelImpl.lambda$setCaretsAndSelections$4(CaretModelImpl.java:460)
	at com.intellij.openapi.editor.impl.CaretModelImpl.doWithCaretMerging(CaretModelImpl.java:419)
	at com.intellij.openapi.editor.impl.CaretModelImpl.setCaretsAndSelections(CaretModelImpl.java:447)
	at com.intellij.openapi.fileEditor.impl.text.TextEditorProvider.setStateImpl(TextEditorProvider.java:261)
	at com.intellij.openapi.fileEditor.impl.text.PsiAwareTextEditorProvider.setStateImpl(PsiAwareTextEditorProvider.java:101)
	at com.intellij.openapi.fileEditor.impl.text.AsyncEditorLoader.setEditorState(AsyncEditorLoader.kt:214)
	at com.intellij.openapi.fileEditor.impl.text.TextEditorImpl.setState(TextEditorImpl.java:163)
	at com.intellij.openapi.fileEditor.impl.FileEditorManagerImpl.restoreEditorState(FileEditorManagerImpl.java:1239)
	at com.intellij.openapi.fileEditor.impl.FileEditorManagerImpl.openFileImpl4Edt(FileEditorManagerImpl.java:1082)
	at com.intellij.openapi.fileEditor.impl.FileEditorManagerExImpl$openFileOnStartup$2.invokeSuspend$lambda$0(FileEditorManagerExImpl.kt:86)
	at com.intellij.openapi.application.TransactionGuardImpl.performActivity(TransactionGuardImpl.java:113)
	at com.intellij.openapi.fileEditor.impl.FileEditorManagerExImpl$openFileOnStartup$2.invokeSuspend(FileEditorManagerExImpl.kt:82)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at com.intellij.openapi.application.TransactionGuardImpl$1.run(TransactionGuardImpl.java:194)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:838)
	at com.intellij.openapi.application.impl.ApplicationImpl$3.run(ApplicationImpl.java:454)
	at com.intellij.openapi.application.impl.FlushQueue.doRun(FlushQueue.java:74)
	at com.intellij.openapi.application.impl.FlushQueue.runNextEvent(FlushQueue.java:114)
	at com.intellij.openapi.application.impl.FlushQueue.flushNow(FlushQueue.java:36)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:318)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:779)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:730)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:724)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:749)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:909)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:756)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$5(IdeEventQueue.java:437)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:787)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$6(IdeEventQueue.java:436)
	at com.intellij.openapi.application.TransactionGuardImpl.performActivity(TransactionGuardImpl.java:105)
	at com.intellij.ide.IdeEventQueue.performActivity(IdeEventQueue.java:615)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$7(IdeEventQueue.java:434)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:838)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:480)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:207)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:128)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:117)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:105)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:92)
Caused by: java.lang.ExceptionInInitializerError
	at java.base/jdk.internal.misc.Unsafe.allocateInstance(Native Method)
	at java.base/java.lang.invoke.DirectMethodHandle.allocateInstance(DirectMethodHandle.java:520)
	at com.intellij.serviceContainer.ComponentManagerImpl.instantiateClass(ComponentManagerImpl.kt:950)
	... 73 more
Caused by: java.lang.NullPointerException: Cannot invoke "java.util.List.get(int)" because "org.antlr.jetbrains.st4plugin.psi.STTokenTypes.TOKEN_ELEMENT_TYPES" is null
	at org.antlr.jetbrains.st4plugin.psi.STTokenTypes.getTokenElementType(STTokenTypes.java:23)
	at org.antlr.jetbrains.st4plugin.highlight.STBraceMatcher.<clinit>(STBraceMatcher.java:16)
	... 76 more

**Comment by jowilkes** [comment]
Problem persists in IntelliJ IDE 2023.1 build #UI-231.8109.175

**Comment by jowilkes** [comment]
Problem persists in IntelliJ IDEA 2023.13 build #UI-231.9161.38

**Comment by rwperrott** [comment]
It'd be nice if this was finally fixed, because I recently restarted work on my Maven plugin for ST4.

**Comment by jowilkes** [comment]
I agree!
Thank you, @FHannes, for starting to work on this.

**Comment by bjansen** [comment]
Fixed by #44
